modules/core/src/main/java/org/apache/ignite/internal/processors/cache/binary/BinaryMetadataVersionInfo.java

```diff
@@ -17,36 +17,55 @@
 package org.apache.ignite.internal.processors.cache.binary;
 
 import java.io.Serializable;
+import org.apache.ignite.IgniteCheckedException;
+import org.apache.ignite.internal.GridKernalContext;
+import org.apache.ignite.internal.Order;
 import org.apache.ignite.internal.binary.BinaryMetadata;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.apache.ignite.plugin.extensions.communication.Message;
 
 /**
  * Wrapper for {@link BinaryMetadata} which is stored in metadata local cache on each node.
  * Used internally to track version counters (see javadoc for {@link MetadataUpdateProposedMessage} for more details).
  * The version refers solely to the internal protocol for updating BinaryMetadata and is unknown externally.
  * It can be updated dynamically from different nodes and threads on the same node.
  */
-final class BinaryMetadataVersionInfo implements Serializable {
+public final class BinaryMetadataVersionInfo implements Serializable, Message {
+    /** Type code. */
+    public static final short TYPE_CODE = 505;
+
     /** */
     private static final long serialVersionUID = 0L;
 
     /** The actual binary metadata. */
-    private final BinaryMetadata metadata;
+    private BinaryMetadata metadata;
+
+    /** Serialized binary metadata. */
+    @Order(0)
+    private byte[] metadataBytes;
```

Comment:
I suggest adding `transient` on this new field to possibly reduce amount of data transferred over the network e.g. on node join.